### PR TITLE
Enh/classif det

### PR DESCRIPTION
**Pull request by stephlazar**
Improves classify_shots of the UHFQC_classifier_detector:

- raises errors when combination of boolean flags does not make sense

- adds option to do ro_corrected_stored_mtx or ro_corrected_seq_cal_mtx. **This means that now you do not have to set the qubit parameter acq_state_prob_mtx to the identity in order to disable RO correction in the detector. The readout correction is not entirely controlled by the two new flags.**

ro_corrected_stored_mtx : corrects with the corresponding qubit matrix stored in acq_state_prob_mtx 

ro_corrected_seq_cal_mtx: corrects with state probability matrix calculated from the averaged calibration states at the end of each sequence.

The changes to the detector function proposed in this pull request were tested both on a virtual setup and on the XLD physical setup.

The changes proposed in this pull request should not break existing code, and users should not need to change their workflow other than the statement above in **bold**.

@chellings @nathlacroix be aware of this pull request. Your feedback is of course welcome.

### CODE REVIEW

File: pycqed/measurement/detector_functions.py [hunk 1]
```diff
@@ -1359,54 +1359,84 @@ def process_data(self, data_raw):
...
 
         return classified_data.T
 
     def classify_shots(self, data, classifier_params_list,
-                       state_prob_mtx_list=None, averaged=False,
-                       thresholded=True):
+                       averaged=True, thresholded=False,
+                       ro_corrected_stored_mtx=False,
+                       ro_corrected_seq_cal_mtx=False,
+                       state_prob_mtx_list=None, **kw):
```
Comment (antsr) [inline pycqed/measurement/detector_functions.py]: Could you add a docstring explaining the meaning of all the parameters of this function.

File: pycqed/measurement/detector_functions.py [hunk 2]
```diff
@@ -1317,14 +1317,14 @@ def prepare(self, sweep_points):
...
         self.UHFQC.qas_0_result_source(self.result_logging_mode_idx)
-        self.UHFQC.qudev_acquisition_initialize(channels=self.channels, 
-                                          samples=self.nr_shots*self.nr_sweep_points,
-                                          averages=1, #for single shot readout
-                                          loop_cnt=int(self.nr_shots),
-                                          mode='rl')
+        self.UHFQC.qudev_acquisition_initialize(
+            channels=self.channels, samples=self.nr_shots*self.nr_sweep_points,
+            averages=1, #for single shot readout
+            loop_cnt=int(self.nr_shots), mode='rl')
```
Comment (antsr) [inline pycqed/measurement/detector_functions.py:1398]: I guess it is not the scope of this PR, but the init here is very similar to the integration_logging_detector. This class should maybe inherit from the logging_det to clean up the code.

**Review by antsr — CHANGES_REQUESTED** [review]
Could you please update the docstring of the class itself and add docstrings to some of the more complex intermediate functions.

This class is really hard to read for me as it is now. Where different parameters are passed in / defined could be improved for this class in general. Example: The split between process_data and classify_shots seems a bit arbitrary. Some of the initial processing is done in process_data, but then rest of it is done in classify shots.

File: pycqed/measurement/detector_functions.py (hunk 1 above)
Comment (stephlazar) [reply pycqed/measurement/detector_functions.py]: I now added explicit docstring.

File: pycqed/measurement/detector_functions.py (hunk 2 above)
Comment (stephlazar) [reply pycqed/measurement/detector_functions.py:1398]: I made this class inherit from the UHFQC_integration_logging_det and cleaned it up a bit.

**Comment by stephlazar** [comment]
@antsr I added explicit docstring, I cleaned up the class and made it inherit from the UHFQC_integration_logging_det.
However, now I think we should first test that everything still works in a measurement on the setup.

**Comment by antsr** [comment]
I'll try and test this on XLD this week to get it merged.

**Comment by stephlazar** [comment]
Sounds good! 👍

**Comment by stephlazar** [comment]
I added even more improvements

- [x] Was tested on virtual setup

- [x] Should still be tested on a real setup

**Review by antsr — APPROVED** [review]
Looks good to me now.

**Pull request merged**